game_events.go

```diff
@@ -186,6 +186,11 @@ func newGameEventHandler(parser *Parser) gameEventHandler {
 }
 
 func (geh gameEventHandler) roundStart(data map[string]*msg.CSVCMsg_GameEventKeyT) {
+
+	// Thrown grenades could not be deleted at the end of the round (if they are thrown at the very end, they never get destroyed)
+	// So we clear them out when a new round start
+	geh.gameState().thrownGrenades = make(map[*common.Player][]*common.Equipment)
```

Comment:
let's move this to `roundOfficiallyEnded` here: https://github.com/markus-wa/demoinfocs-golang/blob/master/game_events.go#L240

That way all these cleanup operations are in the same place 🙂